pyveda/frameworks/batch_generator.py

```diff
@@ -12,13 +14,17 @@ class BaseGenerator():
     Rescale: Boolean. Flag to indicate if data returned from the generator should be rescaled between 0 and 1.
     '''
 
-    def __init__(self, cache, batch_size=32, shuffle=True, rescale=False):
+    def __init__(self, cache, batch_size=32, shuffle=True, rescale=False,
+                    flip_horizontal=False, flip_vertical=False):
         self.cache = cache
         self.batch_size = batch_size
         self.index = 0
         self.shuffle = shuffle
         self.rescale = rescale
         self.on_epoch_end()
+        self.flip_h = flip_horizontal
+        self.flip_v = flip_vertical
+        self._applied_augs(self.flip_h, self.flip_v)
```

Comment:
I think this needs to be on an image by image basis. If this is setup on init then the random switches are applied only once. So every image will be aug'd the same. In the apply_transforms, it should figure out, randomly, which augs to apply to every image. 

Some will be flipped_vertifcally in a batch and others flipper_horizontally and others not touched.

Reply:
well calling this here doesn't do anything at all except return an array that never get's used or dealt with. @martham93 don't the transforms get applied sample by sample? it looks like a loop over it in line 158

Reply:
so  `transforms = self._applied_augs(self.flip_h, self.flip_v)` happens in line 62 of `apply_transforms` and `self.apply_transforms` is called for VedaBase 150 and for VedaStream in 179